<img src="https://coursesandconferences.wellcomeconnectingscience.org/wp-content/themes/wcc_courses_and_conferences/dist/assets/svg/logo.svg" width="300" height="50"> 

# Genetic Analysis of Population-based Association Studies

## 2–6 September 2024, Wellcome Genome Campus, UK

[Wellcome Connecting Science Course Website Link](https://coursesandconferences.wellcomeconnectingscience.org/event/genetic-analysis-of-population-based-association-studies-20240902/) <br /> 
[Course Time table](https://github.com/WCSCourses/Association_Studies_2024/blob/main/Masterplanner%20document%20-%20Timetable.pdf) <br /> 
[Course Informatics Guide](https://github.com/WCSCourses/Association_Studies_2024/blob/main/Course_Informatics_Guide.md)

# Summary

This advanced course aims to give researchers involved in genetic disease studies a firm grounding in the use of the latest statistical methods and software for analysis of genetic association studies. This includes both small-scale disease-specific studies and large-scale collaborative projects including those that can be used for analysis of multiple complex traits such as UK Biobank.

The course will cover both theoretical and practical aspects of the design and analysis of such studies. Each topic will include a lecture followed by a practical session in which state-of-the-art statistical software will be applied to relevant datasets. The practical sessions will illustrate the ideas presented in the lectures. All the software used will be freely available so that skills learned can be applied after the course.

The programme will also include seminars from internationally renowned researchers from the field of complex disease genetics, along with opportunities for participants to discuss their own research projects with course instructors and with each other.

# Programme

The programme will include lecture and computer-based practical sessions covering the following topics:

**Introduction to genetic association studies**

- Overview and history of genetic association studies leading up to and including genome-wide association studies. 

**Basic association analysis and meta-analysis**

- Single marker association tests (frequentist and Bayesian approaches) 
- Calculation of odds ratios and relative risks 
- Logistic regression 
- Meta-analysis (fixed- and random-effects)

**Quality control and population structure**

- Data quality control procedures to avoid the generation of spurious false positives in association studies 
- The confounding effects of population structure on association studies and methods for protecting against these effects 
- Multivariate (principal components) and linear mixed modelling approaches to adjust for population substructure and relatedness in genome-wide association studies

**Haplotype estimation and genotype imputation**

- Methods for genotype imputation using publicly available reference panels 
- Pre-phasing of haplotypes and imputation based on these inferred haplotypes 
- Frequentist and Bayesian methods of testing association at imputed SNPs and indels 
- Quality control for imputed SNPs 
- Meta-analysis using imputed data

**Fine-mapping**

- Methods to identify distinct association signals
- Fine-mapping causal variants and construction of credible sets
- Leveraging diverse populations for fine-mapping through multi-ancestry meta-analysis

**Analysis of rare variants**

- Methods for analysing rare variants from re-sequencing, genotyping and imputation studies via “collapsing approaches”
- Burden and dispersion tests of association

**Mendelian randomisation**

- Concepts behind using genetic variants for causal inference in epidemiology
- Descriptions of the assumptions, limitations and sensitivity analyses of Mendelian randomisation
- Practical session to apply the above and reproduce examples from the literature 

**Introduction to post-GWAS interrogation**

- Approaches to integrate genome-wide association signals with multi-omics to understand disease biology

**Practical Sessions**
Lectures are followed by practical sessions using realistic datasets so that students learn how to apply the theory. Students will use a variety of computer programs during the course including: IMPUTE2, SHAPEIT2, SNPTEST2, META, GCTA, FaST-LMM, PLINK.

## Learning outcomes
On completion of the course, participants can expect to: 

- Understand the general principles, assumptions and basic techniques used in genetic association studies 
- Read and comprehend scientific articles that present results from candidate-gene and genome-wide association studies 
- Analyse genetic data arising from candidate-gene and genome-wide association studies, (including quality control checks, association testing between genotype and phenotype, and post-association fine-mapping of implicated loci to point to likely causal variants, genes and pathways) 
- Perform imputation of variants that have not been directly genotyped, using information from genotyped genetic variants 
- Perform analysis of common and rare variants, accounting for population structure accounting for population structure and family data

**Course Instructors**      

- [Svetlana Cherlin](https://www.newcastle-biostatistics.com/staff/svetlana_cherlin/), Newcastle University, UK
- [Heather Cordell](https://www.staff.ncl.ac.uk/heather.cordell/), Newcastle University, UK
- [Gibran Hemani](https://www.bristol.ac.uk/people/person/Gibran-Hemani-7099528a-3d05-4917-a312-8ac3d8ec5e71/), University of Bristol, UK
- [Andrew Morris](https://research.manchester.ac.uk/en/persons/andrew.morris-5), University of Manchester, UK
- [Gisela Orozco](https://research.manchester.ac.uk/en/persons/gisela.orozco), University of Manchester, UK
- Nik Tzoumas, Newcastle University, UK

**Wellcome Connecting Science Team**    

- [Martin Aslett](https://www.wellcomeconnectingscience.org/person/aslett-martin/), Informatics Manager
- [Vaishnavi Vikas Gangadhar](https://www.wellcomeconnectingscience.org/person/gangadhar-vaishnavi/), Informatics Technical Officer
- [Nicola Stevens](https://www.wellcomeconnectingscience.org/person/stevens-nicola/), Events Organiser

******

## Citing and Re-using Course Material

The course data are free to reuse and adapt with appropriate attribution. All course data in these repositories are licensed under the <a rel="license" href="https://creativecommons.org/licenses/by-nc-sa/4.0/">Attribution-NonCommercial-ShareAlike 4.0 International (CC BY-NC-SA 4.0)</a>. <a rel="license" href="http://creativecommons.org/licenses/by/4.0/"><img alt="Creative Commons Licence" style="border-width:0" src="https://i.creativecommons.org/l/by-nc-sa/4.0/88x31.png" /></a><br /> 

## Interested in attending a course?

Take a look at what courses are coming up at [Wellcome Connecting Science Courses & Conference Website](https://coursesandconferences.wellcomeconnectingscience.org/our-events/).

---

For more information or queries, feel free to contact us via the [Wellcome Connecting Science website](https://coursesandconferences.wellcomeconnectingscience.org).<br /> 


Find us on socials [Wellcome Connecting Science Linktr](https://linktr.ee/eventswcs)

---

[Wellcome Connecting Science GitHub Home Page](https://github.com/WCSCourses) <br /> 

